File: backend/models/event_model.py
from typing import Union
from pydantic import BaseModel
from datetime import datetime, timedelta
from handlers.datetime_handler import DatetimeHandler


from models.short_user_model_response import ShortUserModelResponse
from consts.name_format_event import ONLINE

datetime_handler = DatetimeHandler()


class EventInputModel(BaseModel):
    name: str
    date: int
    format_event: str
    place: dict

    class Config:
        schema_extra = {
            "example": {
                "name": "Вебинар по Angular",
                "date": datetime_handler.now_next_days(2),
                "format_event": ONLINE,
                "place": {"name_platform": "VK", "URL": "https://example"},
            }
        }


class EventInDBModel(EventInputModel):
    key: Union[str, None]
    author: ShortUserModelResponse
    date: int

File: frontend/src/app/shared/models/EventModel.ts
import { ShortUserResponseModel } from "./ShortUserResponseModel";

export const Offline = 'Очно';
export const Online = 'Онлайн';
export class EventModel {
  constructor(
    public name: string,
    public date: number,
    public format_event: string,
    public place: object,
    public key:string,
    public author:ShortUserResponseModel
  ) {}
}
